power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -754,6 +783,16 @@ class Deserializer {
         auto const reordered_attribute_buffers = detail::is_columnar_v<row_or_column_t>
                                                      ? detail::reordered_attribute_buffers(buffer, attributes)
                                                      : std::vector<AttributeBuffer<void>>{};
+        // for columnar buffer
+        // if there is no intersection between the pre-defined attributes and the user provied buffer
+        // and the whole component does not have map
+        // skip the whole component for all scenarios and all elements
+        if constexpr (std::same_as<row_or_column_t, columnar_t>) {
+            if (info.has_attribute_indications && reordered_attribute_buffers.empty()) {
+                component_key_ = "";
+                return;
+            }
+        }
```

Comment:
i personally prefer not to have any early returns in the middle of the function (at start are OK of course), but i understand why it's there.